flang/runtime/pointer.cpp

```diff
@@ -76,9 +76,11 @@ void RTNAME(PointerAssociateLowerBounds)(Descriptor &pointer,
   Terminator terminator{__FILE__, __LINE__};
   std::size_t boundElementBytes{lowerBounds.ElementBytes()};
   for (int j{0}; j < rank; ++j) {
-    pointer.GetDimension(j).SetLowerBound(
-        GetInt64(lowerBounds.ZeroBasedIndexedElement<const char>(j),
-            boundElementBytes, terminator));
+    Dimension &dim{pointer.GetDimension(j)};
+    dim.SetLowerBound(dim.Extent() == 0
```

Comment:
Should this be "<="?  I'm not sure if it's possible for `Extent()` to ever return a negative number.

Reply:
No, `Dimension` extent are always set using `SetBounds` that takes care of setting the extent to zero if the bounds are reversed.